specifyweb/frontend/js_src/lib/components/Header/NotificationRenderers.tsx

```diff
@@ -119,6 +124,67 @@ export const notificationRenderers: IR<
       />
     );
   },
+  'record-merge-starting'(notification) {
+    const tableName = notification.payload.table;
+    const collectionId = parseInt(notification.payload.collection_id);
+    const mergeName = notification.payload.name;
+    const schemaResource = new schema.models.Collection.Resource({
+      id: collectionId,
+    });
+    const [collectionName, setCollectionName] = React.useState<string | null>(
+      ''
+    );
+    schemaResource
+      .fetch()
+      .then((resource) => setCollectionName(resource.get('collectionName')));
+    return (
+      <>
+        {mergingText.mergingHasStarted()}
+        <div className="flex items-center gap-2">
+          <TableIcon label name={tableName} />
+          <p>
+            {collectionName} {'-'} {mergeName}
+          </p>
+        </div>
+      </>
+    );
+  },
+  'record-merge-failed'(notification) {
+    const tableName = notification.payload.table;
+    const id = parseInt(notification.payload.new_record_id);
+    const ids = [JSON.parse(notification.payload.old_record_ids), id];
+    return (
+      <>
+        {mergingText.mergingHasFailed()}
+        <div className="flex items-center gap-2">
+          <TableIcon label name={tableName} />
+          <Link.NewTab
+            href={`/specify/overlay/merge/${tableName}/?${mergingQueryParameter}=${ids}`}
+          >
+            {mergingText.retryMerge()}
+          </Link.NewTab>
+        </div>
+      </>
+    );
+  },
+  'record-merge-succeeded'(notification) {
+    const id = parseInt(notification.payload.new_record_id);
+    const tableName = notification.payload.table;
+    const model = getModel(tableName);
+    const resource =
```

Comment:
do not create a new resource on each render! you muse call useMemo

`FormattedResource` makes a network request to fetch and format the resource every time the resource changes - which it does on every render here